test/E2ETests/Smoke.E2E.Tests/SuiteLifeCycleTests.cs

```diff
@@ -0,0 +1,438 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Linq;
+
+using Microsoft.MSTestV2.CLIAutomation;
+
+namespace MSTestAdapter.Smoke.E2ETests;
+public class SuiteLifeCycleTests : CLITestBase
+{
+    private const string Assembly = "SuiteLifeCycleTestProject.dll";
+
+    public void ValidateTestRunLifecycle_net6()
+    {
+        ValidateTestRunLifecycle("net6.0");
+    }
+
+    public void ValidateTestRunLifecycle_net462()
+    {
+        ValidateTestRunLifecycle("net462");
+    }
+
+    private void ValidateTestRunLifecycle(string targetFramework)
+    {
+        InvokeVsTestForExecution(new[] { targetFramework + "\\" + Assembly }, targetFramework: targetFramework);
+        Verify(RunEventsHandler.PassedTests.Count == 15);  // the inherite class tests called twice
+
+        int noOfassemblyInitCalled = 0;
+        bool isClassCleanupCalled = false;
+        bool isAssemblyCleanupCalled = false;
+
+        foreach (var testMethod in RunEventsHandler.PassedTests)
+        {
+            Verify(testMethod.Outcome == Microsoft.VisualStudio.TestPlatform.ObjectModel.TestOutcome.Passed);
+
+            noOfassemblyInitCalled += testMethod.Messages.Single().Text.Contains("AssemblyInit was called") ? 1 : 0;
+            isClassCleanupCalled |= testMethod.Messages.Single().Text.Contains("ClassCleanup was called");
```

Comment:
Suggestion: here and below, it might be worth to also use int count so that we have consistency and easier refactoring.